packages/labs/src/frameworks/react/create-component.ts

```diff
@@ -0,0 +1,237 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as ReactModule from 'react';
+
+const reservedReactProperties = new Set([
+  'children',
+  'localName',
+  'ref',
+  // TODO(sorvell): why are the properties below included?
+  'style',
+  'className',
+]);
+
+const listenedEvents: WeakMap<
+  Element,
+  Map<string, EventListenerObject>
+> = new WeakMap();
+
+const setEvent = (
```

Comment:
Add a comment on the purpose of this function

Reply:
Done.